### PR TITLE
Move API to using glob patterns

### LINKED ISSUE #10: Report filenames should be configurable
First of all, thanks for mochawesome and this super-handy merge tool!

I have a little improvement suggestion: mochawesome-merge should have a configurable report filename (like mochawesome has env variable `MOCHAWESOME_REPORTFILENAME` and argument `reportFilename`) instead of being hardcoded to '`mochawesome*.json`'.

Currently, if I set env variable `MOCHAWESOME_REPORTFILENAME`, mochawesome-merge will not be able to find the reports.

### PR DESCRIPTION
**Pull request by ojizero**
This allows for further customization of report file names, fixes #10.

It also upgrades officially supported (and tested against) Node version to 10.0.0.

### CODE REVIEW

File: tests/merge.js [hunk 1]
```diff
@@ -15,31 +13,31 @@ describe('merge', () => {
     })
     expect(suites.length).toBe(3)
 
-    expect(suites[0].tests.length).toBe(2)
-    expect(suites[0].passes.length).toBe(2)
-    expect(suites[0].failures.length).toBe(0)
+    expect(suites[0].tests.length).toBe(1)
+    expect(suites[0].passes.length).toBe(0)
+    expect(suites[0].failures.length).toBe(1)
```
Comment (Antontelesh) [inline tests/merge.js]: Why the test assertions have changed?

Comment (ojizero) [reply tests/merge.js]: the ordering of the files returned is different for some reason, if i run sort on the resulted array of files the old assertions would work, but that's an extra operation for just switching some files around an array

**Comment by ojizero** [comment]
@Antontelesh just updated the readme file, sorry completely forgot about it when I made the PR

File: tests/merge.js (hunk 1 above)
Comment (Antontelesh) [reply tests/merge.js]: thank you, makes sense to leave them unsorted

**Pull request merged**